net/pfSense-pkg-Quagga_OSPF/files/usr/local/pkg/quagga_ospfd.inc

```diff
@@ -514,24 +558,86 @@ function quagga_ospfd_plugin_carp($pluginparams) {
 	if (is_array($config['installedpackages']['quaggaospfd']['config'])) {
 		$ospfd_conf = &$config['installedpackages']['quaggaospfd']['config'][0];
 	} else {
+		syslog(LOG_ALERT, "Quagga: CONFIGURATION DOES NOT EXIST!");
 		return null;
 	}
 	/* If there is no properly configured CARP status check IP, then stop */
-	if (!isset($ospfd_conf['carpstatusvid']) || $ospfd_conf['carpstatusvid'] == "none") {
+	if (!isset($ospfd_conf['carpmode']) || $ospfd_conf['carpmode'] == "none") {
+		syslog(LOG_ALERT, "Quagga: CARP failover mode not set.");
 		return null;
 	}
-	list($vhid, $iface) = explode("@", trim($pluginparams['interface']));
-	$friendly = convert_real_interface_to_friendly_interface_name($iface);
-	$vip = get_configured_vip($ospfd_conf['carpstatusvid']);
-	if ($vip['vhid'] != $vhid || $vip['interface'] != $friendly) {
+	/* Verify "quaggadisable" parameters are properly set.*/
+	if ( $ospfd_conf['carpmode'] == "quaggadisable" && (!isset($ospfd_conf['carpstatusvid']) || $ospfd_conf['carpstatusvid'] == "none")) {
+		syslog(LOG_ALERT, "Quagga: Failed checking CARP mode parameters: {$ospfd_conf['carpmode']}");
+		return null;
+	}
+	/* Verify "carpmode" parameters are properly set.*/
+	if ( $ospfd_conf['carpmode'] == "ospfcost" && (!isset($ospfd_conf['carpcostvid'])
+		|| substr($ospfd_conf['carpcostvid'],0,4) == "none")
```

Comment:
All of these conditions should be [indented with four spaces](https://www.netgate.com/docs/pfsense/development/developer-style-guide.html#indent-style)